[feat] Improve flexibility for BOATs integration

MattBuckley-TfN opened the pull request:
# Changes

- Removed `DataBaseHandler` class and instead use `sqlalchemy.Connection` objects for more flexibility
- Added CLI help menu and config example to docs, see [updated docs](https://cafntem--29.org.readthedocs.build/en/29/)

# Tasks

- [x] Have unittests been added (testing individual functions and their edge cases) - No unit tests but integrations tests have been updated
- [x] Has documentation (including user guide) been updated - yes
- [x] Have new dependencies been added (or changed) in relevant `requirements.txt` - added pydantic
- [x] Code linting, tests and other workflows pass

MattBuckley-TfN committed:
[docs] added CLI and config example

Added CLI help to the docs user guide and an example of the query config file.
Removed inherited methods and attributes from generated class documentation.

MattBuckley-TfN committed:
[docs] updated usage and quick start

Filled in quick start and CLI usage sections.
Added announcement explaining the docs are still in development.

MattBuckley-TfN committed:
[docs] parameters to show inherited methods

Added autosummary context variables to enable including inherited methods / attributes.
Also, added list which will show inherited only for some classes.

MattBuckley-TfN committed:
[feat] removed DataBaseHandler class

Removed custom class for running queries and instead passed around `sqlalchemy.Connection` objects to allow for easier integration within BOATs.
Moved `DataBaseHandler.query_to_dataframe` to a function which takes a Connection.
Added missing pydantic dependency.
Ran black / isort on changed modules.

MattBuckley-TfN committed:
[fix] print statements for integration checks

BenTaylor-TfN commented on docs/source/index.rst:
```diff
@@ -7,9 +7,10 @@
+Common analytical framework (CAF) Python package for extracting and analysing NTEM data.
+CAF.ntem provides functionality to load NTEM datasets into a database and
+query the database to produce planning, trip end and car ownership datasets. The
+tools functionality is provided via a command-line interface (CLI) and Python API.
```
```suggestion
Common analytical framework (CAF) Python package for extracting and analysing NTEM data.

NTEM (National Trip End Model) is a UK Department for Transport (DfT) model that forecasts
trip end data up to 2051. More information and the data from this model can be found on 
[data.gov.uk](https://www.data.gov.uk/dataset/11bc7aaf-ddf6-4133-a91d-84e6f20a663e/national-trip-end-model-ntem).

CAF.ntem allows easy querying of this dataset through Python, allowing users to extract
planning, trip end and car ownership datasets. CAF.NTEM provides both a 
command-line interface (CLI) and Python API.

```

BenTaylor-TfN reviewed (commented):
Code changes look good to me, all sensible changes for sqlalchemy. I've added one suggested change for the docs, mostly to describe NTEM and to link to the source so people know where / how to get the data. Not sure if this is mentioned anywhere else?

MattBuckley-TfN committed:
Added NTEM description to README

Co-authored-by: Ben <68055450+BenTaylor-TfN@users.noreply.github.com>

MattBuckley-TfN committed:
[feat] optional zone name column in query output

Added optional parameter to enable including zone name column in output DataFrame's.
Parameter default's to False same as old behaviour.

MattBuckley-TfN committed:
[feat] query output zone property

Added read-only property giving access to the output zone system for a query, needed within BOATs.

MattBuckley-TfN commented:
Added optional parameter to query method which outputs zone name column in addition to zone code

MattBuckley-TfN committed:
[fix] docstring linting

MattBuckley-TfN committed:
[fix] bug with index FrozenList

MattBuckley-TfN committed:
[fix] adding zone name column

MattBuckley-TfN committed:
[fix] pylint false positive
